client/src/main/java/io/pravega/client/connection/impl/AppendBatchSizeTrackerImpl.java

```diff
@@ -27,19 +30,31 @@
  * synchronous writers. Otherwise the batch size is set to the amount of data that will be written in the next
  * {@link #MAX_BATCH_TIME_MILLIS} or half the server round trip time (whichever is less)
  */
-class AppendBatchSizeTrackerImpl implements AppendBatchSizeTracker {
-    private static final int MAX_BATCH_TIME_MILLIS = 100;
-
+public class AppendBatchSizeTrackerImpl implements AppendBatchSizeTracker {
```

Comment:
I have done a new iteration on this branch. The results are much better than before. The only point that needs to be addressed is for small events and 1 segment. In that scenario, for throughput rates between 5K and 300K events per second, the write latency seems few milliseconds higher compared to the PoC. Note that for throughput rates lower and higher than this interval, this branch has a performance equal or better compared to the PoC, which may indicate that this is related to the changes in client batching and not related to any other performance bottleneck.